File: hackerearth/machine-learning/basic-probability.py
def probability(pmb, pab, p1):
    return p1 * ((pmb * (1 - pab) + pab * (1 - pmb)))

def main():
    pmb = float(input())
    pab = float(input())
    p1 = float(input())

    p = probability(pmb, pab, p1)

    print('{:f}'.format(p))

if __name__ == '__main__':
    main()

File: hackerearth/machine-learning/basic-probability.rb
def probability(pmb, pab, p1)
    return p1 * ((pmb * (1 - pab) + pab * (1 - pmb)))
end

def main
    pmb = gets.chomp.to_f
    pab = gets.chomp.to_f
    p1 = gets.chomp.to_f

    p = probability pmb, pab, p1

    puts "%.6f" % p
end

if __FILE__ == $0
    main
end
